### PR TITLE
Get PUID from rps for MSA users 

### PR DESCRIPTION
**Pull request by manikarthikk**
#### What does this PR do? Please provide some context
Get PUID from rps for MSA users 
#### Definition of done:
- [ ] Title of the pull request is clear and informative
- [ ] Add pull request hyperlink to relevant TFS items
- [ ] For large or complex change: schedule an in-person review session
- [ ] This change has appropriate test coverage
- [ ] Get at least two approvals

#### Reminders DURING merge
1. If you're merging from a short-term (feature) branch into a long-term branch (like dev, release, or master) then "Squash and merge" to keep our history clean.
1. If merging from two longterm branches (like cherry picks from upstream, dev to release, etc) then "Create merge commit" to preserve individual commits.

[//]: # ( fyi: This content was heavily inspired by )
[//]: # ( 1 Our team's policies and processes )
[//]: # ( 2 https://github.com/sprintly/sprint.ly-culture/blob/master/pr-template.md )
[//]: # ( 3 The book "The Checklist Manifesto: How to Get Things Right" by Atul Gawande )
[//]: # ( 4 https://github.com/Azure/azure-event-hubs/blob/master/.github/PULL_REQUEST_TEMPLATE.md )

### CODE REVIEW

File: lms/djangoapps/lti_provider/urls.py [hunk 1]
```diff
@@ -5,13 +5,14 @@
 from django.conf import settings
 from django.conf.urls import url
 
-from lti_provider import views
+import lti_provider
```
Comment (eltoncarr) [inline lms/djangoapps/lti_provider/urls.py]: keep import views and align subsequent statements to use views.

File: common/djangoapps/student/views/dashboard.py [hunk 2]
```diff
@@ -547,8 +549,22 @@ def student_dashboard(request):
 
     """
     user = request.user
-    if not UserProfile.objects.filter(user=user).exists():
-        return redirect(reverse('account_settings'))
+    if configuration_helpers.get_value("ENABLE_MSA_MIGRATION"):
+        is_redirection = False
+    try:
+        # Check to see user social entry for this user
```
Comment (eltoncarr) [inline common/djangoapps/student/views/dashboard.py]: Needs a bit of refactor. Move everything under the configuration_helpers.get_value("ENABLE_MSA_MIGRATION") condition.

File: lms/djangoapps/lti_provider/views.py [hunk 3]
```diff
@@ -157,3 +185,55 @@ def parse_course_and_usage_keys(course_id, usage_id):
...
+    except LtiConsumer.DoesNotExist:
+        return HttpResponseForbidden()
+     # Check the OAuth signature on the message
+    if not SignatureValidator(lti_consumer).verify(request):
+        return HttpResponseForbidden()
+     provider = params["provider"]
+    uid = params["uid"]
+    puid = params["puid"]
+     # First verify the mapping is already exist sanity check
+    try:
```
Comment (eltoncarr) [inline lms/djangoapps/lti_provider/views.py]: Put this endpoint behind configuration.

Comment (manikarthikk) [reply lms/djangoapps/lti_provider/views.py]: Done

File: lms/djangoapps/lti_provider/urls.py (hunk 1 above)
Comment (manikarthikk) [reply lms/djangoapps/lti_provider/urls.py]: Done

File: common/djangoapps/student/views/dashboard.py (hunk 2 above)
Comment (manikarthikk) [reply common/djangoapps/student/views/dashboard.py]: Done

**Comment by manikarthikk** [comment]
A typo was fixed with this PR https://github.com/Microsoft/edx-platform/pull/485

**Pull request merged**